<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Document</title>
  </head>
  <body>
    <script>
      //using setTimeout with promies
      function getData(status) {
        return new Promise((resolve, reject) => {
          console.log("getting data! please wait...");
          setTimeout(() => {
            if (status) {
              resolve("Data fetched 100%");
            } else {
              reject("failed to get data");
            }
          }, 3000);
        });
      }
      let onResolve = (result) => console.log(result);
      let onReject = (error) => console.log(error);

      //getData(true).then(onResolve);
      //getData(true).catch(onReject);
      //here we are going to use method chaning
      getData(true).then(onResolve).catch(onReject);

      //example two writing fucntion inside .then and .catch
      function getData1(status) {
        return new Promise((resolve, reject) => {
          console.log("getting data2! please wait...");
          setTimeout(() => {
            if (status) {
              resolve("Data fetched-2 100%");
            } else {
              reject("failed to get data-2");
            }
          }, 4000);
        });
      }
      getData1(true)
        .then((result) => {
          console.log(result);
        })
        .catch((error) => console.log("error:", error));

      //division example
      function division(a, b) {
        return new Promise((resolve, reject) => {
          var c = a / b;
          setTimeout(() => {
            if (c) {
              resolve("your answer " + c);
            } else {
              reject("Failed to calculate");
            }
          }, 5000);
        });
      }
      division(10, 2)
        .then((result) => console.log(result))
        .catch((error) => console.log("error:", error));
    </script>
  </body>
</html>
